src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/SearchEnginesController.php

```diff
@@ -0,0 +1,258 @@
+        $searchEngineForm = $searchEngineFormBuilder->getForm();
+        $searchEngineForm->handleRequest($request);
+
+        try {
+            $result = $searchEngineFormHandler->handle($searchEngineForm);
+
+            if (null !== $result->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_search_engines_index');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Configure/ShopParameters/TrafficSeo/SearchEngines/create.html.twig', [
+            'searchEngineForm' => $searchEngineForm->createView(),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'enableSidebar' => true,
+        ]);
+    }
+
+    /**
+     * Show search engine edit form page and handles its submit.
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
+     *
+     * @param int $searchEngineId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editAction(int $searchEngineId, Request $request): Response
+    {
+        $searchEngineFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.search_engine_form_handler');
+        $searchEngineFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.search_engine_form_builder');
+
+        try {
+            $searchEngineForm = $searchEngineFormBuilder->getFormFor($searchEngineId);
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_search_engines_index');
+        }
+
+        try {
+            $searchEngineForm->handleRequest($request);
+            $result = $searchEngineFormHandler->handleFor($searchEngineId, $searchEngineForm);
+
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_search_engines_index');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            if ($e instanceof SearchEngineNotFoundException) {
+                return $this->redirectToRoute('admin_search_engines_index');
+            }
+        }
+
+        /** @var EditableSearchEngine $editableSearchEngine */
+        $editableSearchEngine = $this->getQueryBus()->handle(new GetSearchEngineForEditing($searchEngineId));
+
+        return $this->render('@PrestaShop/Admin/Configure/ShopParameters/TrafficSeo/SearchEngines/edit.html.twig', [
```

Comment:
Can't we move the template to fit the controller namespace ? (There is an additional level with "TrafficSeo")

Reply:
There is already `@PrestaShop/Admin/Configure/ShopParameters/TrafficSeo/Meta`, what is SEO & URLs page, so this seems like the best place for these templates.